merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/CreateEventPartitionAction.java

```diff
@@ -0,0 +1,32 @@
+package gov.nasa.jpl.aerie.merlin.server.remotes.postgres;
+
+import org.intellij.lang.annotations.Language;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+/*package-local*/ final class CreateEventPartitionAction implements AutoCloseable {
+  private final Connection connection;
+
+  public CreateEventPartitionAction(final Connection connection) throws SQLException {
```

Comment:
🔵 Just for context - I remember the original intent of doing these in triggers was to encapsulate partition management in the database itself. That said, deadlocks are no fun, and I'm certainly more comfortable with Java code than PL/pgSQL, so I'm okay with handling this in Java

Reply:
Is your implication here that we could be triggering the partition creation on the change of status from "pending" to "incomplete" and thus it could remain a triggered function within our schema definitions?

Reply:
I hadn't thought it that far through - but yes, now that you mention it,  I think that would work. I haven't worked with partitions before, so I'm not really sure what to advise here. Maybe we can consult @kroffo or @Twisol or Tung.

Reply:
If you just want the logic centralized in the database, you can always [`CALL`](https://www.postgresql.org/docs/14/sql-call.html) a stored procedure that does the partitioning.

Whether or not this stored procedure is invoked automatically on some change in the database (as a trigger), or manually by the Java client, is an orthogonal concern. 😄